### PR TITLE
#161342802 Implement social authentication

### PR DESCRIPTION
**Pull request by TheoOkafor**
#### What does this PR do?
Implement social authentication

#### Description of Task to be completed?
- Build the social auth component
- Consume the social auth API route

#### How should this be manually tested?
On the command line;
- git clone this repo
- Run `npm start` from the `haven-ah-frontend` directory
- Visit http://localhost:5001

#### What are the relevant pivotal tracker stories?
[#161342802](https://www.pivotaltracker.com/story/show/161342802)

#### Screenshots (if appropriate)>
N/A

### CODE REVIEW

File: src/components/containers/navbar/navbar.scss [hunk 1]
```diff
@@ -1,6 +1,6 @@
 $navbar-background: #fff !important;
 $navbar-box-shadow: 0 1px 3px rgba(0, 0, 0, 0.12), 0 1px 2px rgba(0, 0, 0, 0.24) !important;
-$navbar-transition: all 2.25s ease-in-out;
+$navbar-transition: all 0.5s ease-in-out;
```
Comment (TheoOkafor) [inline src/components/containers/navbar/navbar.scss:3]: Makes Navbar transition from `transparent` to opaque faster

**Comment by ucheg6** [comment]
Great job @TheoOkafor 👍

**Comment by TheoOkafor** [comment]
> Great job @TheoOkafor 👍

Thank you @ucheg6

File: src/components/views/HomePage.jsx [hunk 2]
```diff
@@ -1,12 +1,15 @@
...
 import Navbar from '../containers/navbar/navbar';
 import RecentArticles from '../containers/recentArticles/RecentArticles';
 import FeaturedAuthor from '../containers/featuredAuthor/FeaturedAuthor';
 import TrendingSection from '../trendingsection/TrendingSection';
+import getAndSetUrlToken from '../../utilities/urlTokenHandler';
 
 class HomePage extends PureComponent {
   render() {
+    const { href } = window.location;
+    getAndSetUrlToken(href);
```
Comment (seyi-adeleke) [inline src/components/views/HomePage.jsx:12]: Can't you do this in componentDidmount?

Comment (TheoOkafor) [reply src/components/views/HomePage.jsx:12]: We can.

Comment (seyi-adeleke) [reply src/components/views/HomePage.jsx:12]: so please do it there

Comment (TheoOkafor) [reply src/components/views/HomePage.jsx:12]: I have now. Just realized that I have not pushed it.

**Review by seyi-adeleke — APPROVED** [review]

**Pull request merged**